packages/dx-react-grid-demos/src/demo-sources/grid-virtual-scrolling/scroll-to-row.partial.jsxt

```diff
@@ -0,0 +1,72 @@
+import React, { useState, useRef } from 'react';<%&additionalImports%>
+<%&imports%>
+import {
+  Grid,
+  VirtualTable,
+  TableHeaderRow,
+  Toolbar,
+} from '@devexpress/dx-react-grid-<%&themeName%>';
+import {
+  Plugin,
+  Template,
+  TemplatePlaceholder,
+} from '@devexpress/dx-react-core';
+
+import {
+  generateRows,
+  defaultColumnValues,
+} from '../../../demo-data/generator';
+
+const getRowId = row => row.id;
+
+<%&body%>
+
+export default () => {
+  const [columns] = useState([
+    { name: 'id', title: 'ID' },
+    { name: 'name', title: 'Name' },
+    { name: 'gender', title: 'Gender' },
+    { name: 'city', title: 'City' },
+    { name: 'car', title: 'Car' },
+  ]);
+  const [rows] = useState(generateRows({
+    columnValues: { id: ({ index }) => `id_${99999 - index}`, ...defaultColumnValues },
+    length: 100000,
+  }));
+  const [index, setIndex] = useState(0);
+  const [id, setId] = useState(0);
+  const vtRef = useRef();
+
+  const topRowChange = ({ id: newId, index: newIndex }) => {
+    setIndex(newIndex);
+    setId(newId);
+  };
+
+  const goToRow = (row) => {
+    vtRef.current.scrollToRow(row);
```

Comment:
This method calls a `setState()` function. Why we go our from React basic concepts `controlled/uncontrolled` modes via properties. Why we should call the `setState()` method manually? I think it is not a good idea.

Can we use a pair of properties `topRow` and `onTopRowChange`?

Reply:
We can't use the controlled mode, because the index matches with an interval of height. We cannot identify scroll position by index correctly (scroll is blocked).
We decided to add method, for manual scrolling.